addons/account/models/account_move.py

```diff
@@ -1,349 +1,1869 @@
+                })
+            else:
+                create_method = in_draft_mode and self.env['account.move.line'].new or self.env['account.move.line'].create
+                candidate = create_method({
+                    'name': self.invoice_payment_ref or '/',
+                    'debit': balance < 0.0 and -balance or 0.0,
+                    'credit': balance > 0.0 and balance or 0.0,
+                    'quantity': 1.0,
+                    'amount_currency': -amount_currency,
+                    'date_maturity': date_maturity,
+                    'move_id': self.id,
+                    'currency_id': self.currency_id.id if self.currency_id != self.company_id.currency_id else False,
+                    'account_id': account.id,
+                    'partner_id': self.commercial_partner_id.id,
+                    'exclude_from_invoice_tab': True,
+                })
+            terms_lines_to_keep += candidate
+
+        if in_draft_mode:
+            terms_lines_to_keep._onchange_amount_currency()
+            terms_lines_to_keep._onchange_balance()
+
+        self.line_ids -= lines_map['terms_lines'] - terms_lines_to_keep
+        terms_lines = terms_lines_to_keep
+
+        # Update invoice_payment_ref.
+        self.invoice_payment_ref = terms_lines and terms_lines[0].name or '/'
+
+        # Update the date_maturity.
+        self.invoice_date_due = max_date_maturity
+
+    @api.multi
+    def _recompute_dynamic_lines(self, recompute_all_taxes=False):
+        ''' Recompute all lines that depend of others.
+
+        For example, tax lines depends of base lines (lines having tax_ids set). This is also the case of cash rounding
+        lines that depend of base lines or tax lines depending the cash rounding strategy. When a payment term is set,
+        this method will auto-balance the move with payment term lines.
+
+        :param recompute_all_taxes: Force the computation of taxes. If set to False, the computation will be done
+                                    or not depending of the field 'recompute_tax_line' in lines.
+        '''
+        self.ensure_one()
+
+        lines_map = {
+            'base_lines': self.env['account.move.line'],
+            'tax_lines': self.env['account.move.line'],
+            'terms_lines': self.env['account.move.line'],
+            'rounding_lines': self.env['account.move.line'],
+        }
+
+        # Dispatch lines and pre-compute some aggregated values like taxes.
+        for line in self.line_ids:
+            if line.recompute_tax_line:
+                recompute_all_taxes = True
+                line.recompute_tax_line = False
+                # Ensure the tags are correctly set.
+                if self.type in ('out_refund', 'in_refund'):
+                    tax_repartition_field = 'refund_repartition_line_ids'
+                else:
+                    tax_repartition_field = 'invoice_repartition_line_ids'
```

Comment:
Is a receipt like an invoice regarding the repartition lines then?

Reply:
yes